fl4health/clients/tabular_data_client.py

```diff
@@ -0,0 +1,115 @@
+from logging import INFO
+from pathlib import Path
+from typing import Dict, Sequence
+
+import pandas as pd
+import torch
+from flwr.common.logger import log
+from flwr.common.typing import Config, NDArray, Scalar
+
+from fl4health.clients.basic_client import BasicClient
+from fl4health.feature_alignment.constants import (
+    BINARY,
+    FEATURE_INFO,
+    FORMAT_SPECIFIED,
+    INPUT_DIMENSION,
+    ORDINAL,
+    OUTPUT_DIMENSION,
+)
+from fl4health.feature_alignment.tab_features_info_encoder import TabularFeaturesInfoEncoder
+from fl4health.feature_alignment.tab_features_preprocessor import TabularFeaturesPreprocessor
+from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+from fl4health.utils.metrics import Metric
+
+
+class TabularDataClient(BasicClient):
+    def __init__(
+        self, data_path: Path, metrics: Sequence[Metric], device: torch.device, id_column: str, target_column: str
+    ) -> None:
+        super().__init__(data_path, metrics, device)
+        self.parameter_exchanger = FullParameterExchanger()
+        self.tabular_features_info_encoder: TabularFeaturesInfoEncoder
+        self.tabular_features_preprocessor: TabularFeaturesPreprocessor
+        self.df: pd.DataFrame
+        self.input_dimension: int
+        self.output_dimension: int
+        self.id_column = id_column
+        self.target_column = target_column
+        # The aligned data and targets, which are used to construct dataloaders.
+        self.aligned_features: NDArray
+        self.aligned_targets: NDArray
+
+    def setup_client(self, config: Config) -> None:
+        """
+        Initialize the client by encoding the information of its tabular data
+        and initializing the corresponding TabularFeaturesPreprocessor.
+
+        config[FORMAT_SPECIFIED] indicates whether the server has obtained
+        the source of information to perform feature alignment.
+        If it is True, it means the server has obtained such information
+        (either a priori or by polling a client).
+        So the client will encode that information and use it instead
+        to perform feature preprocessing.
+        """
+        assert FORMAT_SPECIFIED in config.keys()
```

Comment:
`narrow_config_type` will perform a similar assertion, so you don't need to do it here 🙂